pkg/operator/podsecurityreadinesscontroller/classification.go

```diff
@@ -0,0 +1,133 @@
+package podsecurityreadinesscontroller
+
+import (
+	"context"
+	"fmt"
+	"strings"
+
+	securityv1 "github.com/openshift/api/security/v1"
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/util/sets"
+	"k8s.io/klog/v2"
+	psapi "k8s.io/pod-security-admission/api"
+)
+
+var (
+	runLevelZeroNamespaces = sets.New[string](
+		"default",
+		"kube-system",
+		"kube-public",
+		"kube-node-lease",
+	)
+)
+
+func (c *PodSecurityReadinessController) classifyViolatingNamespace(ctx context.Context, conditions *podSecurityOperatorConditions, ns *corev1.Namespace, enforceLevel string) error {
+	if runLevelZeroNamespaces.Has(ns.Name) {
+		conditions.addViolatingRunLevelZero(ns)
+		return nil
+	}
+	if strings.HasPrefix(ns.Name, "openshift") {
+		conditions.addViolatingOpenShift(ns)
+		return nil
+	}
+	if ns.Labels[labelSyncControlLabel] == "false" {
+		conditions.addViolatingDisabledSyncer(ns)
+		return nil
+	}
+
+	isUserViolation, err := c.isUserViolation(ctx, ns, enforceLevel)
+	if err != nil {
+		klog.V(2).ErrorS(err, "Error checking user violations", "namespace", ns.Name)
+		// Transient API server error or temporary resource unavailability (most likely).
+		// Theoretically, psapi parsing errors could occur that retry without hope for recovery.
+		return err
+	}
+
+	if isUserViolation {
+		conditions.addViolatingUserSCC(ns)
+		return nil
+	}
+
+	// Historically, we assume that this is a customer issue, but
+	// actually it means we don't know what the root cause is.
+	conditions.addViolatingCustomer(ns)
+
+	return nil
+}
+
+func (c *PodSecurityReadinessController) isUserViolation(ctx context.Context, ns *corev1.Namespace, label string) (bool, error) {
+	var enforcementLevel psapi.Level
+	switch strings.ToLower(label) {
+	case "restricted":
+		enforcementLevel = psapi.LevelRestricted
+	case "baseline":
+		enforcementLevel = psapi.LevelBaseline
+	case "privileged":
+		// If privileged is violating, something is seriously wrong
+		// but testing against privileged level is pointless (everything passes)
+		klog.V(2).InfoS("Namespace violating privileged level - skipping user check",
+			"namespace", ns.Name)
+		return false, nil
+	default:
+		return false, fmt.Errorf("unknown level: %q", label)
+	}
```

Comment:
I think I might be the one who originally introduced this switch (sorry) but could we make use of `psapi.ParseLevel` instead here?